github/git_refs.go

```diff
@@ -58,13 +60,53 @@ func (s *GitService) GetRef(ctx context.Context, owner string, repo string, ref
 
 	r := new(Reference)
 	resp, err := s.client.Do(ctx, req, r)
-	if err != nil {
+	if _, ok := err.(*json.UnmarshalTypeError); ok {
+		// Multiple refs, means there wasn't an exact match.
+		return nil, resp, nil
+	} else if err != nil {
 		return nil, resp, err
 	}
 
 	return r, resp, nil
 }
 
+// GetRefs fetches a slice of Reference objects for a given Git ref.
+// If a Git ref does not match exactly, GitHub will return all refs that match it as a prefix.
+// Example:
+// feature -> [featureA, featureB]
+// featureA -> [featureA]
+//
+// GitHub API docs: https://developer.github.com/v3/git/refs/#get-a-reference
```

Comment:
We should tweak this comment's formatting. Currently, it will not look good in godoc:

![image](https://cloud.githubusercontent.com/assets/1924134/24569592/d5497010-1634-11e7-8e0f-d5b495a0dc5a.png)

I would suggest formatting it somewhat like this:

```
// GetRefs fetches a slice of Reference objects for a given Git ref.
// If there is an exact match, only that ref is returned.
// If there is no exact match, GitHub returns all refs that start with ref.
// For example:
//
// 	"heads/featureA" -> ["refs/heads/featureA"]                         // Exact match, single ref is returned.
// 	"heads/feature"  -> ["refs/heads/featureA", "refs/heads/featureB"]  // All refs that start with ref.
//
// GitHub API docs: https://developer.github.com/v3/git/refs/#get-a-reference
```

Which looks like this:

![image](https://cloud.githubusercontent.com/assets/1924134/24569828/18738e92-1636-11e7-8e04-b692b302d027.png)